diff --git a/Topology/ArgumentProcessor.php b/Topology/ArgumentProcessor.php
index d5aaf1769eae..27fc62dbf982 100644
--- a/Topology/ArgumentProcessor.php
+++ b/Topology/ArgumentProcessor.php
@@ -1,10 +1,15 @@
 <?php
+
+declare(strict_types=1);
+
 /**
  * Copyright © Magento, Inc. All rights reserved.
  * See COPYING.txt for license details.
  */
 namespace Magento\Framework\Amqp\Topology;

+use InvalidArgumentException;
+
 /**
  * @deprecated 103.0.0
  * see: https://github.com/php-amqplib/php-amqplib/issues/405
@@ -17,22 +22,23 @@ trait ArgumentProcessor
      * @param array $arguments
      * @return array
      */
-    public function processArguments($arguments)
+    public function processArguments($arguments): array
     {
         $output = [];
         foreach ($arguments as $key => $value) {
             if (is_array($value)) {
                 $output[$key] = ['A', $value];
-            } elseif (is_int($value)) {
-                $output[$key] = ['I', $value];
+            } elseif (is_numeric($value)) {
+                $output[$key] = ['I', (int) $value];
             } elseif (is_bool($value)) {
                 $output[$key] = ['t', $value];
             } elseif (is_string($value)) {
                 $output[$key] = ['S', $value];
             } else {
-                throw new \InvalidArgumentException('Unknown argument type ' . gettype($value));
+                throw new InvalidArgumentException('Unknown argument type ' . gettype($value));
             }
         }
+
         return $output;
     }
 }
